src/main/java/org/osc/controller/nsc/api/NeutronSecurityControllerApi.java

```diff
@@ -254,17 +238,26 @@ private void updateInspectionHookProfile(String region, InspectionHook inspectio
             inspectionHook.setHookId(UUID.randomUUID().toString());
         }
 
-        UpdatePort updatedPort = Port.updateBuilder()
-                .profile(InspectionHook.updateBindingProfile(inspectionHook, inspectedPort.getProfile())).build();
-        neutronPortApi.update(inspectionHook.getInspectedPortId(), updatedPort);
+        try {
+            getOs().useRegion(region);
+            ImmutableMap<String, Object> bindingProfile = InspectionHook.updateBindingProfile(inspectionHook, inspectedPort.getProfile());
+            getOs().networking().port().update(inspectedPort.toBuilder().profile(bindingProfile).build());
+        } finally {
+            getOs().removeRegion();
+        }
+
     }
 
-    private Port getPortOrThrow(PortApi neutronPortApi, String portId, PortType type)
-            throws NetworkPortNotFoundException {
-        Port port = neutronPortApi.get(portId);
+    private Port getPortOrThrow(String region, String portId, PortType type) throws NetworkPortNotFoundException {
+        getOs().useRegion(region);
+        Port port;
+        try {
+            port = getOs().networking().port().get(portId);
+        } finally {
+            getOs().removeRegion();
+        }
         if (port == null) {
```

Comment:
if port is null or the action response is not 200...we want to throw the exception